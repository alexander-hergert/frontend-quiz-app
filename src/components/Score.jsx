import React from "react";
import Link from "next/link";

const Score = ({ score, icon, title }) => {
  let bgColor = "";
  if (title === "HTML") {
    bgColor = "bg-html";
  } else if (title === "CSS") {
    bgColor = "bg-css";
  } else if (title === "JavaScript") {
    bgColor = "bg-js";
  } else if (title === "Accessibility") {
    bgColor = "bg-accessibility";
  } else if (title === "React") {
    bgColor = "bg-react";
  } else if (title === "Node.js") {
    bgColor = "bg-nodejs";
  } else if (title === "SQL") {
    bgColor = "bg-sql";
  } else if (title === "Git / Bash") {
    bgColor = "bg-git";
  }

  return (
    <section
      className="flex justify-between max-md:my-4 max-md:w-[327px] md:w-[640px] max-xl:flex-col xl:w-[1160px] max-md:h-[457px] 
        md:h-[638px] xl:h-[456px]  max-md:gap-[40px] md:gap-[64px] xl:gap-[140px]"
    >
      <h1 className="max-md:text-[2rem] xl:text-[3rem] max-md:h-[88px] xl:h-[136px] w-full">
        Quiz completed, You scored...
      </h1>
      <div className="flex flex-col gap-[32px] md:w-[640px] xl:w-[564px] h-[512px]">
        <div className="flex flex-col justify-center items-center bg-primary max-md:rounded-[12px] xl:rounded-[24px] h-[388px]">
          <div className="flex items-center h-[56px] gap-[24px]">
            <div
              className={`w-[56px] h-[56px] flex justify-center items-center ${bgColor} rounded`}
            >
              <img src={icon} alt="topic" className="w-[40px] h-[40px]" />
            </div>
            <h2 className="text-2xl">{title}</h2>
          </div>
          <div className="h-[196px]">
            <p className="text-[6rem]">{score}</p>
            <p className="text-accent">out of 10</p>
          </div>
        </div>
        <Link
          href={"/"}
          className="btn max-md:h-[56px] h-[92px] w-full max-md:rounded-[12px] xl:rounded-[24px] text-xl
           text-white border-none bg-nav hover:bg-[#d394fa]"
        >
          Play Again
        </Link>
      </div>
    </section>
  );
};

export default Score;
