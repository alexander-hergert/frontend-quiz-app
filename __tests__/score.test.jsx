import "@testing-library/jest-dom";
import { render, screen } from "@testing-library/react";
import Score from "../src/components/Score";

describe("Score", () => {
  it("renders correctly", () => {
    render(
      <Score
        score={0}
        icon={"/assets/images/icon-html.svg"}
        title={"HTML"}
        numberOfQuestions={10}
      />
    );
    expect(screen.getByTestId("score")).toBeInTheDocument();
    expect(screen.getByTestId("score")).toHaveTextContent("0");
    expect(screen.getByTestId("icon")).toBeInTheDocument();
    expect(screen.getByTestId("icon")).toHaveAttribute(
      "src",
      "/assets/images/icon-html.svg"
    );
    expect(screen.getByTestId("title")).toBeInTheDocument();
    expect(screen.getByTestId("title")).toHaveTextContent("HTML");
    expect(screen.getByTestId("numberofquestions")).toBeInTheDocument();
    expect(screen.getByTestId("numberofquestions")).toHaveTextContent(
      "out of 10"
    );
  });
  //background color
  it("renders background color correctly", () => {
    render(
      <Score
        score={0}
        icon={"/assets/images/icon-html.svg"}
        title={"HTML"}
        numberOfQuestions={10}
      />
    );
    expect(screen.getByTestId("background")).toHaveClass("bg-html");
  });
  //test Link to route to /
  it("renders link correctly", () => {
    render(
      <Score
        score={0}
        icon={"/assets/images/icon-html.svg"}
        title={"HTML"}
        numberOfQuestions={10}
      />
    );
    expect(screen.getByTestId("link")).toHaveAttribute("href", "/");
  });

  //snapshot test
  it("renders unchanged", () => {
    const { container } = render(
      <Score
        score={0}
        icon={"/assets/images/icon-html.svg"}
        title={"HTML"}
        numberOfQuestions={10}
      />
    );
    expect(container).toMatchSnapshot();
  });
});
