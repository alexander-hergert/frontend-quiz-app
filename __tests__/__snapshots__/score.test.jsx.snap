// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Score renders unchanged 1`] = `
<div>
  <section
    class="flex justify-between max-md:my-4 max-md:w-[327px] md:w-[640px] max-xl:flex-col xl:w-[1160px] max-md:h-[457px]  md:h-[638px] xl:h-[456px] max-md:gap-[40px] md:gap-[64px] xl:gap-[140px]"
  >
    <h1
      class="max-xl:text-[2rem] xl:text-[3rem] max-md:h-[88px] xl:h-[136px] w-full"
    >
      Quiz completed, You scored...
    </h1>
    <div
      class="flex flex-col gap-[32px] h-[512px]"
    >
      <div
        class="flex flex-col justify-center md:w-[640px] xl:w-[564px] items-center bg-primary max-md:rounded-[12px] md:rounded-[24px] h-[388px]"
      >
        <div
          class="flex items-center h-[56px] gap-[24px]"
        >
          <div
            class="w-[56px] h-[56px] flex justify-center items-center bg-html rounded"
            data-testid="background"
          >
            <img
              alt="topic"
              class="w-[40px] h-[40px]"
              data-testid="icon"
              src="/assets/images/icon-html.svg"
            />
          </div>
          <h2
            class="text-2xl"
            data-testid="title"
          >
            HTML
          </h2>
        </div>
        <div
          class="h-[196px]"
        >
          <p
            class="text-[6rem]"
            data-testid="score"
          >
            0
          </p>
          <p
            class="text-accent"
            data-testid="numberofquestions"
          >
            out of 
            10
          </p>
        </div>
      </div>
      <a
        class="btn max-md:h-[56px] h-[92px] w-full max-md:rounded-[12px] xl:rounded-[24px] text-xl text-white border-none bg-nav hover:bg-[#d394fa]"
        data-testid="link"
        href="/"
      >
        Play Again
      </a>
    </div>
  </section>
</div>
`;
